### PR TITLE
fix: page change in saved insights preserves filters

### LINKED ISSUE #23672: Cannot use insight filters beyond the first page
## Bug description

2024-07-12 at 12 04 02

## How to reproduce

1. View the insights list at https://us.posthog.com/insights (make sure you have enough insights for the pagination to work)
2. Apply a filter (such as `only show trends`)
3. Try to view the 2nd page
4. Notice that all insight filters have been removed...

## Additional context

https://posthoghelp.zendesk.com/agent/tickets/15609

### Debug info

```shell
Session: https://us.posthog.com/project/sTMFPsFhdP1Ssg/replay/0190a65c-4cee-7540-a3e8-1fda71cb3a4d?t=134 
Admin: http://go/adminOrgUS/0180f4b1-65c0-0000-7f11-eb8e661b782b (Project: 10085)
Sentry: http://go/sentryUS/10085
```

### PR DESCRIPTION
**Pull request by anirudhpillai**
## Problem
Fixes #23672 

## Changes
Think at some point the `urls.savedInsights()` got changed from `/insights` to `/projects/:id/insights` so currently changing filters on the saved insights (product analytics page) wouldn't update the URL and the filters weren't preserved when the page changes (as the URL wouldn't have any filters other than page id)

Prev behaviour (URL doesn't update to store the updated filters)
image

After Changes (URL updates to have the filters)
image

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?
N/A

## How did you test this code?
Tested on Local UI by changing filters

### CODE REVIEW

File: frontend/src/scenes/saved-insights/savedInsightsLogic.ts [hunk 1]
```diff
@@ -321,17 +321,10 @@ export const savedInsightsLogic = kea<savedInsightsLogicType>([
                   }
               ]
             | void => {
-            if (router.values.location.pathname === urls.savedInsights()) {
```
Comment (anirudhpillai) [inline frontend/src/scenes/saved-insights/savedInsightsLogic.ts:324]: Noticed we don't check the URL in other places, maybe should updated `urls.savedInsights()` as well? (then can keep this check)

Comment (thmsobrmlr) [reply frontend/src/scenes/saved-insights/savedInsightsLogic.ts:324]: I think this line is here for instances when the `savedInsightsLogic` is mounted, but you are not actually on the saved insights page. We should probably adapt this line to still check the pathname and just ignore the url params there (url starts with or something).

Comment (Twixes) [reply frontend/src/scenes/saved-insights/savedInsightsLogic.ts:324]: The cleanest way to do this check is to check `sceneLogic.values.activeScene`, like here: https://github.com/PostHog/posthog/blob/bab882c5fec15f9b6af72c252f52276ad4df774f/frontend/src/scenes/insights/insightSceneLogic.tsx#L235

**Comment by skoob13** [comment]
@anirudhpillai, the changes look good to me! Filters definitely work after the first page now.

There is a small problem though: when you navigate from the first page to the second page and press the browser's back button, you stay on the same page. If you press the back button again, it will actually navigate you back. See the GIF below.



I would merge it as is because this issue is super annoying and fix the problem above separately.

**Review by skoob13 — APPROVED** [review]

**Comment by anirudhpillai** [comment]
Thanks @skoob13 for noticing that bug!
It seems we update the url twice while pagination as the `usePagination` hook [already does it](https://github.com/PostHog/posthog/blob/master/frontend/src/lib/lemon-ui/PaginationControl/usePagination.ts#L15)

So I've in turn [removed the `onForward/onBackward` logic](https://github.com/PostHog/posthog/pull/23722/files#diff-8993faccbfcd0bbb467cc93c0aee6930faa02290e66650db94ed39071b4a5a43L230-L237) so that we don't modify the URL twice

**Pull request merged**